File: Frontend/src/app/Dtos/CourseSearchResultDto.ts
import { TagDto } from "./TagDto";

export interface CourseSearchResultDto {
  id: number;
  name: string;
  description: string;
  thumbnailUrl?: string | null;
  instructor: string;
  estimatedDurationInHours: number;
  lessonCount?: number | null;
  oneTimePrice?: number | null;
  currency?: string | null;
  category?: string | null;
  tags: TagDto[];
  matchType: string;
}

File: Backend/src/main/java/com/example/e_learning_system/Dto/CourseDtos/CourseSearchResultDto.java
package com.example.e_learning_system.Dto.CourseDtos;

import com.example.e_learning_system.Config.Category;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CourseSearchResultDto {
    private int id;
    private String name;
    private String description;
    private String thumbnailUrl;
    private String instructor;
    private int estimatedDurationInHours;
    private Integer lessonCount;
    private BigDecimal oneTimePrice;
    private String currency;
    private Category category;
    private Set<TagDto> tags;
    private String matchType; // "title", "description", "tag", "category"
}
